Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -493,12 +505,18 @@ void SetupIDs()
                     // force 0 for prefabs
                     sceneId = 0;
                     //Debug.Log(name + " @ scene: " + gameObject.scene.name + " sceneid reset to 0 because CurrentPrefabStage=" + PrefabStageUtility.GetCurrentPrefabStage() + " PrefabStage=" + PrefabStageUtility.GetPrefabStage(gameObject));
-                    // NOTE: might make sense to use GetPrefabStage for asset
-                    //       path, but let's not touch it while it works.
 #if UNITY_2020_1_OR_NEWER
                     string path = PrefabStageUtility.GetCurrentPrefabStage().assetPath;
+
+                    // Try GetPrefabStage if empty path
+                    if (string.IsNullOrEmpty(path))
+                        path = PrefabStageUtility.GetPrefabStage(gameObject).assetPath;
```

Comment:
shouldn't this be a separate PR?
and the comments only explain that it tries getprefabstage if empty path.
but it doesn't explain why path might be empty and why getprefabstage might be the answer.

please explain

Reply:
Separate PR #2781